### ISSUE TITLE
Upgrade to Python 3.10

### ISSUE THREAD

**Issue by achimnol** [lablup/backend.ai#300]
Python 3.10.0 is live now.
Let's upgrade our codebase (the 21.03/21.09 branch and the main branch for 22.03) to use Python 3.10 for future backportability.

Currently known blockers:
 - python/cpython#28850 (expected to land in 3.10.1)
 - aio-libs/janus#358
 - actions/setup-python#299
 - ~~syrusakbary/aiodataloader#16 (merged but not released, the maintainer seems to have gone...)~~ forked and released as [aiodataloader-ng](https://github.com/lablup/aiodataloader-ng)
 - achimnol/aiotools#28
 - python/mypy#10191 (expected to land in 0.940)
 - grpc/grpc#28387
   - Mitigated by #391

Not blockers but considerable stuffs for seamless development experience:
 - vim-python/python-syntax#80

**Comment by achimnol** [comment]
Currently GitHub Action Runner supports only 3.10.0, not 3.10.1 that we need. On hold.

**Comment by achimnol** [comment]
GitHub Action Runner now provides 3.10.1.
Also fixed some new typing errors found by mypy 0.920 update.

**Comment by achimnol** [comment]
Forked aiodataloader as [aiodataloader-ng](https://github.com/lablup/aiodataloader-ng).

**Comment by achimnol** [comment]
Released our own version of aiodataload-ng: https://pypi.org/project/aiodataloader-ng/0.2.1/

**Comment by kyujin-cho** [comment]
newly investigated upstream issues:
- grpc/grpc#28387

**Comment by achimnol** [comment]
> newly investigated upstream issues:
> 
> * [[Python][macOS] 3.10 wheel is mislabeled as universial2 grpc/grpc#28387](https://github.com/grpc/grpc/issues/28387)

Currently the known workaround is to execute the followings in the target venv:
```shell
pip install --no-binary :all: grpcio --ignore-installed
pip install --no-binary :all: grpcio-tools --ignore-installed
```

**Comment by achimnol** [comment]
After #417, we now work fine on Pytho 3.10.5 and now all external dependencies have upgraded to provide 3.10-compatible binary wheels.  The vim syntax support could be workarounded using NeoVim and treesitter.
